### PR TITLE
Move test utils in the src dir

### PR DESCRIPTION
**Pull request by andreivladbrg**
Same rationale as [here](https://github.com/PaulRBerg/prb-math/pull/206)

### Changelog:

refactor: move test utils in src dir
build: install prb test as npm package
build: remove prb test git module
build: use correct path in update-precompiles script
ci: install node js dependencies in build test coverage job
chore: update remappings accordingly

### CODE REVIEW

**Comment by andreivladbrg** [comment]
I see that the coverage has significantly decreased. This is because there are more contracts in `src` which don't have to be tested.

Is there a way to remove a path (~src/test/) from the coverage report?

File: package.json [hunk 1]
```diff
@@ -9,6 +9,9 @@
   "bugs": {
     "url": "https://github.com/PaulRBerg/prb-proxy/issues"
   },
+  "dependencies": {
```
Comment (PaulRBerg) [inline package.json:12]: Has to be `devDependencies` and also has to be listed in `peerDependencies`

**Review by PaulRBerg — CHANGES_REQUESTED** [review]
Thanks Andrei, just like in PRBMath, let's install Forge Std as a Node.js dependency.

> Is there a way to remove a path (~src/test/) from the coverage report?

Yes. See the `codecov.yml` file and the [Ignoring Paths](https://docs.codecov.com/docs/ignoring-paths) docs.

**Comment by andreivladbrg** [comment]
Addressed everything in my latest commits.

I see that the slither job has failed and I am not sure why, is it because we need to add `pnpm install` step?

**Comment by PaulRBerg** [comment]
Thanks!

> I see that the slither job has failed

Dunno, need to investigate but cannot at the moment

**Comment by andreivladbrg** [comment]
Closing in favor of https://github.com/PaulRBerg/prb-proxy/pull/179

**Pull request closed without merging**